Assets/Scripts/Entities/EnemyAI2.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyAI2 : MonoBehaviour
+{
+    public float shootSpeed;
+    public Transform shootPos;
+    public GameObject missile;
+    public Collider2D col;
+    public GameObject enemy;
+    public int ammo;
+    public Bounds missilePosition;
+    public GameObject canvas;
+    // Start is called before the first frame update
+    void Start()
+    {
+        enemy = GameObject.Find(Constants.ENEMY2_PREFAB);
+        StartCoroutine(Spawn(enemy, 5.0f));
+        enemy.SetActive(true);
+        StartCoroutine(AttackFan.Shoot(missile, 1, ammo, shootSpeed, shootPos));
```

Comment:
Not entirely a fan of having an entire new AI script just for a different attack pattern, is there some way we can dynamically assign different attack patterns to the same EnemyAI script?

Reply:
The GameObject.SendMessage function is a pretty good substitute